sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java

```diff
@@ -426,6 +424,9 @@ private String isCancellationProneChannel(Timestamps timestamps, Instant current
             final long readSuccessRecency = Duration.between(timestamps.lastChannelReadTime(), currentTime).toNanos();
 
             if (readSuccessRecency >= this.nonRespondingChannelReadDelayTimeLimitInNanos) {
+
+                final Optional<RntbdContext> rntbdContext = requestManager.rntbdContext();
```

Comment:
how moving it here helps?
`requestManager can still be null here, right?

Reply:
`requestManager` is checked for nullability at the beginning of `isHealthyWithFailureReason`. Moving it within at least guarantees the channel has seen wire reads which would imply RNTBD context negotiation has happened (in-line with other health check gates).